flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityContext.java

```diff
@@ -0,0 +1,218 @@
+		System.setProperty("java.security.auth.login.config", "");
+
+		// establish the UGI login user
+		UserGroupInformation.setConfiguration(config.hadoopConf);
+		UserGroupInformation loginUser;
+		if(UserGroupInformation.isSecurityEnabled() && config.keytab != null && !Preconditions.isNullOrEmpty(config.principal)) {
+
+			String keytabPath = (new File(config.keytab)).getAbsolutePath();
+			// login with keytab
+			UserGroupInformation.loginUserFromKeytab(config.principal, keytabPath);
+
+			loginUser = UserGroupInformation.getLoginUser();
+
+			// supplement with any available tokens
+			String fileLocation = System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
+			if(fileLocation != null) {
+				/*
+				 * Use reflection API since the API semantics are not available in Hadoop1 profile. Below APIs are
+				 * used in the context of reading the stored tokens from UGI.
+				 * Credentials cred = Credentials.readTokenStorageFile(new File(fileLocation), config.hadoopConf);
+				 * loginUser.addCredentials(cred);
+				*/
+				try {
+					Method readTokenStorageFileMethod = Credentials.class.getMethod("readTokenStorageFile",
+																File.class, org.apache.hadoop.conf.Configuration.class);
+					Credentials cred = (Credentials) readTokenStorageFileMethod.invoke(null,new File(fileLocation),
+																							config.hadoopConf);
+					Method addCredentialsMethod = UserGroupInformation.class.getMethod("addCredentials",
+																						Credentials.class);
+					addCredentialsMethod.invoke(loginUser,cred);
+				} catch(NoSuchMethodException e) {
+					LOG.warn("Could not find method implementations in the shaded jar. Exception: {}", e);
+				}
+			}
+		} else {
+			// login with current user credentials (e.g. ticket cache)
+			try {
+				//Use reflection API to get the login user object
+				//UserGroupInformation.loginUserFromSubject(null);
+				Method loginUserFromSubjectMethod = UserGroupInformation.class.getMethod("loginUserFromSubject", Subject.class);
+				Subject subject = null;
+				loginUserFromSubjectMethod.invoke(null,subject);
+			} catch(NoSuchMethodException e) {
+				LOG.warn("Could not find method implementations in the shaded jar. Exception: {}", e);
+			}
+
+			loginUser = UserGroupInformation.getLoginUser();
+			// note that the stored tokens are read automatically
+		}
+
+		if(UserGroupInformation.isSecurityEnabled() && !UserGroupInformation.getLoginUser().hasKerberosCredentials()) {
+			LOG.error("Hadoop Security is enabled but current login user does not have Kerberos Credentials");
+			throw new RuntimeException("Hadoop Security is enabled but current login user does not have Kerberos Credentials");
+		}
+
+		installedContext = new SecurityContext(loginUser);
+	}
+
+	/**
+	 * Inputs for establishing the security context.
+	 */
+	public static class SecurityConfiguration {
+
+		Configuration flinkConf = null;
```

Comment:
Not necessary to initialize field variables with `null` (they are by default null if not initialized).